Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+* for `GET`: 400, 401, 403, 500, 503
+* for `GET/{subscriptionId}`: 400, 401, 403, 404, 500, 503
+* dor `DELETE`: 400, 401, 403, 404, 500, 503
+
+_Termination for resource-based subscription_
+
+3 scenarios are possibles (business conditions may apply):
+
+* case1: subscriptionExpireTime has been provided in the request and reached. The operator in this case has to terminate the subscription.
+* case2: subscriber requests a DELETE for a subscription. 
+* case3: subscription ended at operator request. 
+
+It could be useful to provide a mechanism to inform subscriber for case3 (and probably case1). In this case a specific event type could be used.
+
+
+_Subscription example_
+
+Request:
+
+```json
+curl -X 'POST' \
+  'http://localhost:9091//device-status/v0/subscriptions/roamingStatus' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d
+{
+  "notificationUrl": "https://application-server.com",
+  "notificationAuthToken": "c8974e592c2fa383d4a3960714",
+  "subscriptionDetail": {
+    "ueId": {,
+      "ipv4Addr": "192.168.0.1"
+    },
+    "uePort": 5060
+  },
+  "subscriptionExpireTime": "2023-03-31T00:00:00.000Z"
+}
+```
+
+response:
+
+```json
+201 Created
+
+{
+  "notificationUrl": "https://application-server.com",
+  "notificationAuthToken": "c8974e592c2fa383d4a3960714",
+  "subscriptionDetail": {
+    "ueId": {
+      "ipv4Addr": "192.168.0.1"
+    },
+    "uePort": 5060
+  },
+  "subscriptionExpireTime": "2023-03-31T00:00:00.000Z",
+  "subscriptionId": "456g899g",
+  "type": "ROAMING_STATUS",
+  "startedAt": "2023-03-17T16:02:41.314Z"
+}
+```
+
+
+
+
+### 12.2 Event notification
+
+The notification endpoint is used by the API server to notify the API consumer that an event occured.
+
+Note: The notification is the message posted on listener side. We describe the notification message in the CAMARA OAS but it could confusing because this endpoint should be implemented on the business API consumer side. This notice should be explicited mentionned in all CAMARA API documentation featuring notification.
+
+Only Operation POST is provided for `notifications` and the expected response code is `204`.
+
+For consistence between CAMARA API an uniform `notifications` model must be used:
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| subscriptionId | string | subscription identifier - could be valued for Resource-based subscription | optional |
+| eventType | string | type of event as defined in each CAMARA API. The event type are defined in UPPER_SNAKE_CASE| mandatory |
+| eventTime | string - datetime | date time when the event was notified by the issuer- valued in the request. This is not the date when the event occurred | mandatory |
+| eventDetail | object | A detailed event structure depending on the eventType | mandatory |
+
+
+Specific eventType "SUBSCRIPTION_ENDS" is defined to inform listener about subscrition termination.
```

Comment:
Would like to understand how this is intended to work

Reply:
The idea behind is the following:
Suppose that as API server for any reason you want to stop sending notification for an event type to a subscriber (you sell this feature to send notification and you got a problem with your customer ). We need to inform the app developer about this service notification ends. It means that we have an event SUBSCRIPTION_ENDS and we send it to the callback URL where we send the 'nominal' event. We should probably have an uniform structure for this event details.